posthog/models/cohort/dependencies.py

```diff
@@ -0,0 +1,157 @@
+import logging
+
+from django.core.cache import cache
+from django.db.models.signals import post_delete, post_save
+from django.dispatch import receiver
+
+from posthog.models.cohort.cohort import Cohort
+
+logger = logging.getLogger(__name__)
+DEPENDENCY_CACHE_TIMEOUT = 24 * 60 * 60  # 24 hours
+TEAM_CACHE_TIMEOUT = 23 * 60 * 60  # 23 hours - expires before individual entries
+
+
+def extract_cohort_dependencies(cohort) -> set[int]:
+    """
+    Extract cohort dependencies from the given cohort.
+    """
+    dependencies = set()
+
+    for prop in cohort.properties.flat:
+        if prop.type == "cohort" and isinstance(prop.value, int):
+            dependencies.add(prop.value)
+
```

Comment:
This doesn't handle the legacy `groups` format.

```suggestion
    for prop in cohort.properties.flat:
        if prop.type == "cohort" and isinstance(prop.value, int):
            dependencies.add(prop.value)

for group in cohort.groups:
    if not isinstance(group, dict):
        continue

    properties = group.get("properties", [])
    if isinstance(properties, list):
        for prop in properties:
            if isinstance(prop, dict) and prop.get("type") == "cohort":
                value = prop.get("value")
                if isinstance(value, int):
                    dependencies.add(value)
```